src/gui/uithememanager.cpp

```diff
@@ -140,5 +165,43 @@ QString UIThemeManager::getIconPath(const QString &iconId) const
         return path;
     }
 #endif
-    return IconProvider::instance()->getIconPath(iconId);
+
+    const auto iter = m_iconMap.find(iconId);
+    QString iconPath = (iter != m_iconMap.end()) ? iter->toString() : QString{};
+    QString iconIdPattern = iconId;
+    while (iconPath.isEmpty() && !iconIdPattern.isEmpty()) {
+        const int sepIndex = iconIdPattern.indexOf('.');
+        iconIdPattern = "*" + ((sepIndex == -1) ? "" : iconIdPattern.right(iconIdPattern.size() - sepIndex));
+        const auto patternIter = m_iconMap.find(iconIdPattern);
+        if (patternIter != m_iconMap.end())
+            iconPath = patternIter->toString();
+        else
+            iconIdPattern.remove(0, 2); // removes "*."
+    }
+
+    if (iconPath.isEmpty()) {
+        LogMsg(tr("Can't resolve icon id - %1").arg(iconId), Log::WARNING);
```

Comment:
Since missing ID isn't really error case (as I said before this case can occur when new ID is introduced in some qBittorrent version but theme is outdated yet) you should fallback to default theme icon.

The real error cases are:
1. Empty filename
2. File doesn't exist

IMO, we should reject such invalid theme at all (and show message to the user).

I also think you can check theme for these errors once it is loaded (since loaded theme is immutable).

Reply:
Do you mean by  - for every `iconId` which doesn't exist in iconMap, try finding that in defaultIconMap which means - 
1. Load both default iconconfig and custom iconconfig (if custom is enabled) in the constructor
2. Reject custom theme if "real error case" exists, (should I check the default Theme for errors too, but only show the error)
3. In `getIconPath`, for every `iconId` which doesn't exist in iconMap, try finding that in defaultIconMap

Reply:
>for every iconId which doesn't exist in iconMap, try finding that in defaultIconMap

Generally, Yes.
But I think there is a better algorithm:
1. Load default iconconfig. Check it and show errors (or maybe don't check it at all?).
2. Load custom iconconfig. Check it and reject if "real error case" exists
3. Join both iconconfigs (add unique and replace existing values in default config with the ones from custom config.
4. Use resulting icon config.

Reply:
While checking should I load the icon directly i.e store the icons in something like `std::map<QString, QIcon>`,